File: src/main/java/ru/serdtsev/homemoney/moneyoper/MoneyOperRepo.java
package ru.serdtsev.homemoney.moneyoper;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.PagingAndSortingRepository;
import ru.serdtsev.homemoney.balancesheet.BalanceSheet;
import ru.serdtsev.homemoney.moneyoper.model.MoneyOper;
import ru.serdtsev.homemoney.moneyoper.model.MoneyOperStatus;

import java.sql.Date;
import java.util.UUID;
import java.util.stream.Stream;

public interface MoneyOperRepo extends PagingAndSortingRepository<MoneyOper, UUID> {
  Stream<MoneyOper> findByBalanceSheet(BalanceSheet balanceSheet);
  Page<MoneyOper> findByBalanceSheetAndStatus(BalanceSheet balanceSheet, MoneyOperStatus status, Pageable pageable);
  Page<MoneyOper> findByBalanceSheetAndStatusAndId(BalanceSheet balanceSheet, MoneyOperStatus status, UUID id, Pageable pageable);
  Page<MoneyOper> findByBalanceSheetAndStatusAndPerformed(BalanceSheet balanceSheet, MoneyOperStatus status, Date performed, Pageable pageable);
  Stream<MoneyOper> findByBalanceSheetAndStatusAndPerformed(BalanceSheet balanceSheet, MoneyOperStatus status, Date performed);
}

File: src/main/java/ru/serdtsev/homemoney/moneyoper/MoneyOperRepo.kt
package ru.serdtsev.homemoney.moneyoper

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import ru.serdtsev.homemoney.balancesheet.BalanceSheet
import ru.serdtsev.homemoney.moneyoper.model.MoneyOper
import ru.serdtsev.homemoney.moneyoper.model.MoneyOperStatus
import java.time.LocalDate
import java.util.*
import java.util.stream.Stream

interface MoneyOperRepo : PagingAndSortingRepository<MoneyOper, UUID> {
    fun findByBalanceSheet(balanceSheet: BalanceSheet): List<MoneyOper>
    fun findByBalanceSheetAndStatus(balanceSheet: BalanceSheet, status: MoneyOperStatus, pageable: Pageable): Page<MoneyOper>
    fun findByBalanceSheetAndStatusAndId(balanceSheet: BalanceSheet, status: MoneyOperStatus, id: UUID,
            pageable: Pageable): Page<MoneyOper>
    fun findByBalanceSheetAndStatusAndPerformed(balanceSheet: BalanceSheet, status: MoneyOperStatus, performed: LocalDate,
            pageable: Pageable): Page<MoneyOper>
    fun findByBalanceSheetAndStatusAndPerformed(balanceSheet: BalanceSheet, status: MoneyOperStatus,
            performed: LocalDate): List<MoneyOper>
    fun findByBalanceSheetAndStatusAndPerformedGreaterThan(balanceSheet: BalanceSheet, status: MoneyOperStatus,
            performed: LocalDate): List<MoneyOper>
}